libraries/chain/database.cpp

```diff
@@ -2005,6 +2014,94 @@ void database::apply_operation(const operation& op)
    notify_post_apply_operation( note );
 }
 
+void database::process_tx_bandwidth(const signed_transaction& trx) {
+   account_bandwidth_object trx_bandwidth_data;
+
+   trx_bandwidth_data.total_bandwidth = fc::raw::pack_size(trx);
+   trx_bandwidth_data.total_ops_count = trx.operations.size();
+
+   bool fee_free_op_present = false;
+   bool paid_op_present     = false;
+   for (const operation& op : trx.operations) {
```

Comment:
How expensive is this processing? This code can heavily influence overall performance. Perhaps, can this inner cycle be moved inside the cycle on lines 1996-1999 to avoid going through the cycle twice? Also, some optimization that the bandwidth data are updated only once after all block is processed shall be considered.